### PR TITLE
square brackets and colons as first word in commit delimiters

### PR DESCRIPTION
**Pull request by clover-bt**
Accommodate messages that begin with:

```
[BAR-104] fourth
[BAR-105][section] fifth
[BAR-106]: sixth
```

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/jiraext/view/FirstWordOfCommitStrategy.java [hunk 1]
```diff
@@ -69,13 +69,23 @@ public FirstWordOfCommitStrategy()
         String msg = change.getMsg();
         String firstWordOfTicket;
         firstWordOfTicket = msg.substring(0, (msg.contains(" ") ? StringUtils.indexOf(msg, " ") : msg.length()));
+        firstWordOfTicket = firstWordOfTicket.substring(0, (firstWordOfTicket.contains("]") ? StringUtils.indexOf(firstWordOfTicket, "]") : firstWordOfTicket.length()));
```
Comment (dalvizu) [inline src/main/java/org/jenkinsci/plugins/jiraext/view/FirstWordOfCommitStrategy.java]: first word of ticket is meant to hold the first string of characters not containing a space (or the entire commit message, if it contains a space)

File: src/main/java/org/jenkinsci/plugins/jiraext/view/FirstWordOfCommitStrategy.java [hunk 2]
```diff
@@ -69,13 +69,23 @@ public FirstWordOfCommitStrategy()
...
+        firstWordOfTicket = firstWordOfTicket.substring(0, (firstWordOfTicket.contains(":") ? StringUtils.indexOf(firstWordOfTicket, ":") : firstWordOfTicket.length()));
 
         for (String validJiraPrefix : Config.getGlobalConfig().getJiraTickets())
         {
             if (firstWordOfTicket.endsWith(":"))
             {
                 firstWordOfTicket = firstWordOfTicket.substring(0, firstWordOfTicket.length() - 1);
             }
+            if (firstWordOfTicket.startsWith("["))
```
Comment (dalvizu) [inline src/main/java/org/jenkinsci/plugins/jiraext/view/FirstWordOfCommitStrategy.java]: This followed a bad example (line #77) -- this only supported removing a single character. Really what we want is to see if the first 'word' (string of characters without whitespace) contained a valid JIRA ticket. Tickets have the form `<prefix>-<number>` - we should be matching that to regex

**Comment by clover-bt** [comment]
Thanks, @dalvizu . The code did what needed to be done, generally following the existing code. Good to know about the openness to regex. Can provide an update. :+1:

**Comment by clover-bt** [comment]
@dalvizu Please see https://github.com/jenkinsci/jira-ext-plugin/pull/15 for a regex-based approach.

**Pull request merged**